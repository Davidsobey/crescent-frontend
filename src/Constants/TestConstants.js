const TestConstants = {
  CREATE_REQUEST: 'CREATE_TEST_REQUEST',
  CREATE_SUCCESS: 'CREATE_TEST_SUCCESS',
  CREATE_FAILURE: 'CREATE_TEST_FAILURE',

  GETALL_REQUEST: 'GETALL_TEST_REQUEST',
  GETALL_SUCCESS: 'GETALL_TEST_SUCCESS',
  GETALL_FAILURE: 'GETALL_TEST_FAILURE',

  LOADTEST_REQUEST: 'LOADTEST_TEST_REQUEST',
  LOADTEST_SUCCESS: 'LOADTEST_TEST_SUCCESS',
  LOADTEST_FAILURE: 'LOADTEST_TEST_FAILURE',

  LOADTESTQUESTIONS_REQUEST: 'LOADTESTQUESTIONS_TEST_REQUEST',
  LOADTESTQUESTIONS_SUCCESS: 'LOADTESTQUESTIONS_TEST_SUCCESS',
  LOADTESTQUESTIONS_FAILURE: 'LOADTESTQUESTIONS_TEST_FAILURE',

  MARKTESTQUESTION_REQUEST: 'MARKTESTQUESTION_TEST_REQUEST',
  MARKTESTQUESTION_SUCCESS: 'MARKTESTQUESTION_TEST_SUCCESS',
  MARKTESTQUESTION_FAILURE: 'MARKTESTQUESTION_TEST_FAILURE',

  LOADQUESTION_SUCCESS: 'LOADQUESTION_TEST_SUCCESS',
  LOADQUESTION_SUCCES: 'LOADQUESTION_TEST_SUCCES',
  LOADQUESTION_SUCCE: 'LOADQUESTION_TEST_SUCCE',
};

export const test = {
  id: 4,
  name: 'Example Test',
  moduleID: 2,
  totalMarks: 100,
  module: null,
  questions: null,
  questionIds: null,
  enrolments: null,
};

export const questionsObj = [
  {
    id: 1,
    testId: 4,
    courseId: 2,
    userId: 2,
    score: 0,
    questionId: 2,
    marked: true,
    answerGivenId: 3,
    question: {
      title: 'Is this another sample?',
      allocatedMarks: 10,
      testId: 4,
      test: null,
      questionOptions: [
        {
          title: 'No it is not',
          isAnswer: false,
          questionId: 2,
          id: 3,
        },
        {
          title: 'Yes it is',
          isAnswer: true,
          questionId: 2,
          id: 4,
        },
      ],
      id: 2,
    },
    enrolmentTest: null,
    answerGiven: null,
  },
  {
    id: 2,
    testId: 4,
    courseId: 2,
    userId: 2,
    score: 10,
    questionId: 1,
    marked: true,
    answerGivenId: 1,
    question: {
      title: 'This is a sample?',
      allocatedMarks: 10,
      testId: 4,
      test: null,
      questionOptions: [
        {
          title: 'Yes it is',
          isAnswer: true,
          questionId: 1,
          id: 1,
        },
        {
          title: 'No it is not',
          isAnswer: false,
          questionId: 1,
          id: 2,
        },
      ],
      id: 1,
    },
    enrolmentTest: null,
    answerGiven: {
      title: 'Yes it is',
      isAnswer: true,
      questionId: 1,
      question: {
        title: 'This is a sample?',
        allocatedMarks: 10,
        testId: 4,
        test: null,
        questionOptions: [
          {
            title: 'No it is not',
            isAnswer: false,
            questionId: 1,
            id: 2,
          },
        ],
        id: 1,
      },
      id: 1,
    },
  },
  {
    id: 3,
    testId: 4,
    courseId: 2,
    userId: 2,
    score: 0,
    questionId: 2,
    marked: true,
    answerGivenId: 3,
    question: {
      title: 'Is this another sample?',
      allocatedMarks: 10,
      testId: 4,
      test: null,
      questionOptions: [
        {
          title: 'No it is not',
          isAnswer: false,
          questionId: 2,
          id: 3,
        },
        {
          title: 'Yes it is',
          isAnswer: true,
          questionId: 2,
          id: 4,
        },
      ],
      id: 2,
    },
    enrolmentTest: null,
    answerGiven: null,
  },
  {
    id: 4,
    testId: 4,
    courseId: 2,
    userId: 2,
    score: 10,
    questionId: 1,
    marked: true,
    answerGivenId: undefined,
    question: {
      title: 'This is a sample?',
      allocatedMarks: 10,
      testId: 4,
      test: null,
      questionOptions: [
        {
          title: 'Yes it is',
          isAnswer: true,
          questionId: 1,
          id: 1,
        },
        {
          title: 'No it is not',
          isAnswer: false,
          questionId: 1,
          id: 2,
        },
      ],
      id: 1,
    },
    enrolmentTest: null,
    answerGiven: {
      title: 'Yes it is',
      isAnswer: true,
      questionId: 1,
      question: {
        title: 'This is a sample?',
        allocatedMarks: 10,
        testId: 4,
        test: null,
        questionOptions: [
          {
            title: 'No it is not',
            isAnswer: false,
            questionId: 1,
            id: 2,
          },
        ],
        id: 1,
      },
      id: 1,
    },
  },
  {
    id: 5,
    testId: 4,
    courseId: 2,
    userId: 2,
    score: 0,
    questionId: 2,
    marked: true,
    answerGivenId: 3,
    question: {
      title: 'Is this another sample?',
      allocatedMarks: 10,
      testId: 4,
      test: null,
      questionOptions: [
        {
          title: 'No it is not',
          isAnswer: false,
          questionId: 2,
          id: 3,
        },
        {
          title: 'Yes it is',
          isAnswer: true,
          questionId: 2,
          id: 4,
        },
      ],
      id: 2,
    },
    enrolmentTest: null,
    answerGiven: null,
  },
  {
    id: 6,
    testId: 4,
    courseId: 2,
    userId: 2,
    score: 10,
    questionId: 1,
    marked: true,
    answerGivenId: undefined,
    question: {
      title: 'This is a sample?',
      allocatedMarks: 10,
      testId: 4,
      test: null,
      questionOptions: [
        {
          title: 'Yes it is',
          isAnswer: true,
          questionId: 1,
          id: 1,
        },
        {
          title: 'No it is not',
          isAnswer: false,
          questionId: 1,
          id: 2,
        },
      ],
      id: 1,
    },
    enrolmentTest: null,
    answerGiven: {
      title: 'Yes it is',
      isAnswer: true,
      questionId: 1,
      question: {
        title: 'This is a sample?',
        allocatedMarks: 10,
        testId: 4,
        test: null,
        questionOptions: [
          {
            title: 'No it is not',
            isAnswer: false,
            questionId: 1,
            id: 2,
          },
        ],
        id: 1,
      },
      id: 1,
    },
  },
];

export const question = {
  id: 2,
  testId: 4,
  courseId: 2,
  userId: 2,
  score: 0,
  questionId: 2,
  marked: true,
  answerGivenId: 3,
  question: {
    title: 'Is this another sample?',
    allocatedMarks: 10,
    testId: 4,
    test: null,
    questionOptions: [
      {
        title: 'No it is not',
        isAnswer: false,
        questionId: 2,
        id: 3,
      },
      {
        title: 'Yes it is',
        isAnswer: true,
        questionId: 2,
        id: 4,
      },
    ],
    id: 2,
  },
  enrolmentTest: null,
  answerGiven: null,
};

export default TestConstants;
