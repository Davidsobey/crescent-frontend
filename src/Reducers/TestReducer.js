import TestConstants, {
  questionsObj,
  question,
} from '../Constants/TestConstants';

function TestReducer(state = {}, action) {
  switch (action.type) {
    case TestConstants.CREATE_REQUEST:
      return {
        ...state,
        test: action.testName,
      };
    case TestConstants.CREATE_SUCCESS:
      return {
        ...state,
        test: action.testName,
      };
    case TestConstants.CREATE_FAILURE:
      return { ...state };

    case TestConstants.GETALL_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case TestConstants.GETALL_SUCCESS:
      return { ...state, tests: action.tests };
    case TestConstants.GETALL_FAILURE:
      return {
        ...state,
        modules: {},
      };
    case TestConstants.LOADTEST_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case TestConstants.LOADTEST_SUCCESS:
      return {
        ...state,
        test: action.test,
      };
    case TestConstants.LOADTEST_FAILURE:
      return {
        ...state,
        modules: {},
      };

    case TestConstants.LOADTESTQUESTIONS_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case TestConstants.LOADTESTQUESTIONS_SUCCESS: {
      const ques = action.questions.sort((a, b) => a.id - b.id);
      return { ...state, questions: ques };
    }
    case TestConstants.LOADTESTQUESTIONS_FAILURE:
      return {
        ...state,
        modules: {},
      };

    case TestConstants.MARKTESTQUESTION_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case TestConstants.MARKTESTQUESTION_SUCCESS: {
      const newData = state.questions.map((ques) => {
        if (ques.questionId === action.payload.id) {
          return { ...ques, answerGivenId: action.payload.answerId };
        }
        return ques;
      });
      return { ...state, response: 'Question Marked', questions: newData };
    }
    case TestConstants.MARKTESTQUESTION_FAILURE:
      return {
        ...state,
        error: action.error,
      };
    case TestConstants.LOADQUESTION_SUCCESS:
      return {
        ...state,
        question: action.question,
      };
    case TestConstants.CHANGE_ANSWER:
      return {
        ...state,
        question: {
          ...state.question,
          answerGivenId: action.value,
        },
      };
    case TestConstants.LOADQUESTION_SUCCES:
      return {
        ...state,
        questions: questionsObj,
      };
    case TestConstants.LOADQUESTION_SUCCE:
      return {
        ...state,
        question,
      };
    default:
      return state;
  }
}

export default TestReducer;
