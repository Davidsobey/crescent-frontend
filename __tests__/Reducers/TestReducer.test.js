// Reducer to be tested
import TestReducer from '../../src/Reducers/TestReducer';
import TestConstants from '../../src/Constants/TestConstants';

describe('TestReducer', () => {
  test('CREATE_SUCCESS', () => {
    const action = {
      type: TestConstants.CREATE_SUCCESS,
      testName: '111',
      newTestId: 1,
    };

    expect(TestReducer(undefined, action)).toMatchSnapshot();
  });

  test('GETALL_SUCCESS', () => {
    const action = {
      type: TestConstants.GETALL_SUCCESS,
      tests: [],
    };

    expect(TestReducer(undefined, action)).toMatchSnapshot();
  });

  test('LOAD_TEST_SUCCESS', () => {
    const action = {
      type: TestConstants.LOAD_TEST_SUCCESS,
      test: {id:1, name:'111'},
    };

    expect(TestReducer(undefined, action)).toMatchSnapshot();
  });

  test('LOAD_TEST_QUESTIONS_SUCCESS', () => {
    const action = {
      type: TestConstants.LOAD_TEST_QUESTIONS_SUCCESS,
      questions: [],
    };

    expect(TestReducer(undefined, action)).toMatchSnapshot();
  });

  test('MARK_TEST_QUESTION_SUCCESS', () => {
    const action = {
      type: TestConstants.MARK_TEST_QUESTION_SUCCESS,
      payload: {id:1, answerId:1},
    };

    const state = {
      questions: [],
    };

    expect(TestReducer(state, action)).toMatchSnapshot();
  });

  test('SUBMIT_TEST_SUCCESS', () => {
    const action = {
      type: TestConstants.SUBMIT_TEST_SUCCESS,
    };

    expect(TestReducer(undefined, action)).toMatchSnapshot();
  });

  test('EDIT_TEST_SUCCESS', () => {
    const action = {
      type: TestConstants.EDIT_TEST_SUCCESS,
    };

    expect(TestReducer(undefined, action)).toMatchSnapshot();
  });

  test('LOADTEST_SUCCESS', () => {
    const action = {
      type: TestConstants.LOADTEST_SUCCESS,
      tests: [],
    };

    expect(TestReducer(undefined, action)).toMatchSnapshot();
  });
});