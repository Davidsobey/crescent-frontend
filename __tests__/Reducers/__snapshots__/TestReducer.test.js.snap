// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestReducer CREATE_SUCCESS 1`] = `
Object {
  "creating": false,
  "newTestId": 1,
  "test": "111",
}
`;

exports[`TestReducer EDIT_TEST_SUCCESS 1`] = `
Object {
  "test": undefined,
  "test_editing": false,
}
`;

exports[`TestReducer GETALL_SUCCESS 1`] = `
Object {
  "loading": false,
  "tests": Array [],
}
`;

exports[`TestReducer LOAD_TEST_QUESTIONS_SUCCESS 1`] = `
Object {
  "questions": Array [],
}
`;

exports[`TestReducer LOAD_TEST_SUCCESS 1`] = `
Object {
  "test": Object {
    "id": 1,
    "name": "111",
  },
  "test_loading": false,
}
`;

exports[`TestReducer LOADTEST_SUCCESS 1`] = `
Object {
  "loading": false,
  "tests": Array [],
}
`;

exports[`TestReducer MARK_TEST_QUESTION_SUCCESS 1`] = `
Object {
  "questions": Array [],
  "response": "Question Marked",
}
`;

exports[`TestReducer SUBMIT_TEST_SUCCESS 1`] = `Object {}`;
